File: backend/config/environment.js
export const environments = {
  development: {
    baseUrl: 'http://localhost:3000',
    apiUrl: 'http://localhost:3000/api'
  },
  production: {
    baseUrl: 'https://marketplace-backend-jv7c.onrender.com',
    apiUrl: 'https://marketplace-backend-jv7c.onrender.com/api'
  }
};

export const currentEnvironment = process.env.NODE_ENV || 'development';

export const config = environments[currentEnvironment]; 

File: frontend/lib/config/environment.dart
import 'package:flutter/foundation.dart';

enum Environment {
  development,
  production,
}

class EnvironmentConfig {
  static const Environment _currentEnvironment = Environment.development;  // Set to production

  static const Map<Environment, String> _baseUrls = {
    Environment.development: 'http://10.0.2.2:3000',  // For Android emulator
    Environment.production: 'https://marketplace-backend-jv7c.onrender.com',
  };

  static const Map<Environment, String> _apiUrls = {
    Environment.development: 'http://10.0.2.2:3000/api',
    Environment.production: 'https://marketplace-backend-jv7c.onrender.com/api',
  };

  static String get baseUrl => _baseUrls[_currentEnvironment]!;
  static String get apiUrl => _apiUrls[_currentEnvironment]!;
  static bool get isDevelopment => _currentEnvironment == Environment.development;
  static bool get isProduction => _currentEnvironment == Environment.production;

  // Helper method to get the current environment
  static Environment get currentEnvironment => _currentEnvironment;
} 